### PR TITLE
Make node builder.

### PR DESCRIPTION
**Pull request by hakonhall**
While trying to add another field to Node I came upon the multitude of places
the constructor is called. This is simplified by using a builder.

The builder is internal to Node for now. If wanted/needed, it can be made
public (especially tests may be interested - I have touched as few as
possible).

### CODE REVIEW

**Comment by hakonhall** [comment]
@bratseth do you want to review this? Otherwise I'll send it to @mpolden (related to hardware divergence reporting)

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java [hunk 1]
```diff
@@ -45,37 +46,29 @@
...
-        Objects.requireNonNull(id, "A node must have an ID");
-        requireNonEmpty(ipAddresses, "A node must have at least one valid IP address");
-        requireNonEmptyString(hostname, "A node must have a hostname");
-        requireNonEmptyString(parentHostname, "A parent host name must be a proper value");
-        Objects.requireNonNull(flavor, "A node must have a flavor");
-        Objects.requireNonNull(status, "A node must have a status");
-        Objects.requireNonNull(state, "A null node state is not permitted");
-        Objects.requireNonNull(allocation, "A null node allocation is not permitted");
-        Objects.requireNonNull(history, "A null node history is not permitted");
-        Objects.requireNonNull(type, "A null node type is not permitted");
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java]: I think the contracts should be kept here as the `Node` class should have the final say in what's allowed in its fields or not.

Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java]: Agree. It also provides a simple to find and unambiguous answer to the question "what values can this field have?".

Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java]: Done

**Comment by mpolden** [comment]
Merge after freeze lifted.

**Pull request merged**